src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -47,8 +50,12 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
 
             AuthenticationResult authResult;
 
+            bool skipCache = _clientParameters.ForceRefresh || 
+                (!string.IsNullOrEmpty(AuthenticationRequestParameters.Claims) && 
+                string.IsNullOrEmpty(_clientParameters.AccessTokenHashToRefresh));
```

Comment:
I don't understand the interaction between claims and `AccessTokenHashToRefresh`. I think we have 2 choices: 

1. Do not allow WithClaims and this API to be used toghether. OR
2. If both are set, bypass cache if claims AND a precise cache hit occurs.